internal/driver/incominglistener.go

```diff
@@ -8,47 +8,58 @@ package driver
 
 import (
 	"encoding/json"
+	"fmt"
 
 	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
 	"github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
 
 	"github.com/eclipse/paho.mqtt.golang"
 )
 
+const (
+	name = "name"
+	cmd  = "cmd"
+)
+
 func onIncomingDataReceived(client mqtt.Client, message mqtt.Message) {
 	var data map[string]interface{}
 	json.Unmarshal(message.Payload(), &data)
 
-	if !checkDataWithKey(data, "name") || !checkDataWithKey(data, "cmd") {
+	nameVal, ok := data[name]
+	if !ok {
+		driver.Logger.Warnf("[Incoming listener] Incoming reading ignored, reading data `%v` should contain the field `%s` to indicate the device name", data, name)
```

Comment:
The should all be errors, not warnings. They need to be logged when user has log level set to only `ERROR`

Reply:
Modified. Show error logs when the incoming reading is invalid.